modules/effects/arrow-effects/src/main/kotlin/arrow/effects/typeclasses/Fiber.kt

```diff
@@ -0,0 +1,47 @@
+package arrow.effects.typeclasses
+
+import arrow.Kind
+import arrow.effects.IO
+
+/**
+ * [Fiber] represents the pure result of an [Async] data type
+ * being started concurrently and that can be either joined or canceled.
+ *
+ * You can think of fibers as being lightweight threads, a fiber being a
+ * concurrency primitive for doing cooperative multi-tasking.
+ */
+interface Fiber<F, A> {
+
+  /**
+   * Returns a new task that will await for the completion of the
+   * underlying fiber, (asynchronously) blocking the current run-loop
+   * until that result is available.
+   */
+  val join: Kind<F, A>
+
+  /**
+   * Triggers the cancellation of the fiber.
+   *
+   * @returns a task that trigger the cancellation upon evaluation.
+   */
+  val cancel: Kind<F, Unit>
```

Comment:
We already got `CancelToken<F>` ongoing in `Bracket` PR. Should we wait to merge that one and update this to use the typealias ? (this is ` def cancel: CancelToken[F]` in cats).

Reply:
`type CancelToken[F[_]] = F[Unit]` in cats. Anyhow, I am fine with waiting until `Bracket` is merged and do it in 1 shot.

Reply:
Yep that's what I meant about the typealias. Let's wait if there's no hurry for a merge.